addons/mrp/models/mrp_production.py

```diff
@@ -78,10 +78,11 @@ def _get_default_is_locked(self):
     product_uom_category_id = fields.Many2one(related='product_id.uom_id.category_id')
     product_uom_qty = fields.Float(string='Total Quantity', compute='_compute_product_uom_qty', store=True)
     picking_type_id = fields.Many2one(
-        'stock.picking.type', 'Operation Type', copy=True, readonly=False,
+        'stock.picking.type', 'Operation Type', copy=True, readonly=True,
         compute='_compute_picking_type_id', store=True, precompute=True,
         domain="[('code', '=', 'mrp_operation'), ('company_id', '=', company_id)]",
-        required=True, check_company=True, index=True)
+        required=True, check_company=True, index=True,
+        states={'draft': [('readonly', False)]})
```

Comment:
Don't use state. The ORM don't manage it very well with compute (and since we clean all the compute/onchange in saas-15.4 it won't work). Instead you need to adapt the views.

Reply:
I can't actually edit the view, since the issue comes from studio. We should prevent this field from being edited when not in draft mode. Maybe I can add a check in the write method, like this: https://github.com/odoo/odoo/pull/92862#discussion_r888783881

@Whenrow What do you think ?

Reply:
The readonly management for field we do not control makes no sense as the user may always set `readonly=False` in his view. 
I'll close the PR as the improvement is not that useful